sample/machineclass.yaml

```diff
@@ -0,0 +1,32 @@
+# Sample machine-class object
+apiVersion: "cluster.k8s.io/v1alpha1"
+kind: MachineClass
+metadata:
+  name: small
+  namespace: foo
+providerConfig:
+  apiVersion: "gceproviderconfig/v1alpha1"
+  kind: "GCEProviderConfig"
+  project: "$GCLOUD_PROJECT"
+  zone: "us-central1-f"
+  machineType: "n1-standard-2"
+  image: "projects/ubuntu-os-cloud/global/images/family/ubuntu-1604-lts"
+
+---
+
+# Sample machine object
+apiVersion: cluster.k8s.io/v1alpha1
+kind: Machine
+metadata:
+  name: test-machine 
+  namespace: foo
+  labels:
+    test-label: test-label 
+spec:
+  providerConfig:
+    valueFrom:
+      machineClass: 
+        provider: gcp
```

Comment:
I know this is an example, but can provider be defined in the machine class object above?  With the way it was defined above, can provider be provided in either machine class or machine objects?

Reply:
I have kept `provider` at machine-object level, keeping in mind the usecase where we might not want to fetch the complete MachineClass when MachineObject is seen/updated. In the future iterations, we can think if it's needed at any other layers as well to be populated explicitly.

Reply:
Inline providerConfig will still work as it is currently, classes will only be a parallel mechanism for users to inject the providerConfig via reference. Above comment was explicitly for `provider` field at MachineObject layer, and not the existence of classes as such :)